docs/ops/deployment/yarn.md

```diff
@@ -0,0 +1,203 @@
+
+
+### Resource Allocation Behavior
+
+A JobManager submitted in YARN session mode will wait for job submissions before starting to request and allocate TaskManagers. The other deployment modes are immediately requesting the required TaskManagers from YARN.
+
+The memory configurations for JobManager and TaskManager processes will be respected by the YARN implementation. The number of reported VCores is by default equal to the number of configured slots per TaskManager. The [`yarn.containers.vcores`]({% link ops/config.md %}#yarn-containers-vcores) allows overwriting the number of vcores with a custom value. In order for this parameter to work you should enable CPU scheduling in your cluster.
+
+Failed containers (including the JobManager) are replaced by YARN. The maximum number of JobManager container restarts is configured via `yarn.application-attempts` (default 1). The YARN Application will fail once all attempts are exhausted.
+
+
+### High-Availability on YARN
+
+For enabling HA on YARN, the `yarn.application-attempts` need to be greater 1, and a [high availability service needs to be configured]().
+
+### Supported Hadoop versions.
+
+Flink on YARN is compiled against Hadoop 2.4.1, which is also the earliest version we are supporting. 
+
+For providing Flink with the required Hadoop dependencies, we recommend setting the `HADOOP_CLASSPATH` environment variable. 
+If that is not possible, the dependencies can also be put into the `lib/` folder of Flink. Flink also offers pre-bundled Hadoop fat jars for placing them in the `lib/` folder, in the "Downloads / Additional Components" section of the website. These pre-bundled fat jars are shaded to avoid dependency conflicts with common libraries.
+
+The Flink community is not testing the YARN integration against these pre-bundled jars. 
+
+
+
+### Running Flink on YARN behind Firewalls
+
+Some YARN clusters use firewalls for controlling the network traffic between the cluster and the rest of the network.
+In those setups, Flink jobs can only be submitted to a YARN session from within the cluster's network (behind the firewall).
+If this is not feasible for production use, Flink allows to configure a port range for its REST endpoint, used for the client-cluster communication. With this range configured, users can also submit jobs to Flink crossing the firewall.
+
+The configuration parameter for specifying the REST endpoint port is the following:
+
+ * `rest.bind-port`
+
+This configuration option accepts single ports (for example: "50010"), ranges ("50000-50025"), or a combination of
+both ("50010,50011,50020-50025,50050-50075").
+
+Please make sure that the configuration option `rest.port` has not been specified, because it has precedence over `rest.bind-port` and accepts no ranges.
```

Comment:
```suggestion
Please make sure that the configuration option [rest.port]({% link ops/config.md %}#rest-port) has not been specified, because it has precedence over `rest.bind-port` and accepts no ranges.
```

I added the link for `rest.port` here as well.